op-node/rollup/types.go

```diff
@@ -254,6 +267,77 @@ func (cfg *Config) CheckL2GenesisBlockHash(ctx context.Context, client L2Client)
 	return nil
 }
 
+// dependencySetActivation stores sorted timestamp-dependencies pairs
+type dependencySetActivation struct {
+	timestamp uint64
+	chainIDs  []*big.Int
+}
+
+var (
+	// Store activations in sorted order
+	dependencySetActivations []dependencySetActivation
+	// Track latest activated dependency update position
+	latestActivationIndex int = -1
+)
+
+// IsDependencySetUpdate checks for any new chain dependencies that activate between
+// the previous and next block timestamps. Returns nil if no new dependencies activate,
+// or a slice of chain IDs for newly activated dependencies.
+func (cfg *Config) IsDependencySetUpdate(previousBlockTimestamp, nextBlockTimestamp uint64) []*big.Int {
+	if cfg.ClusterConfig == nil {
+		panic("missing cluster config")
+	}
+
+	// Initialize activations on first call
+	if dependencySetActivations == nil {
+		// Convert map to array for more efficient lookups
+		dependencySetActivations = make([]dependencySetActivation, 0, len(cfg.ClusterConfig.DependencySet))
+		for ts, deps := range cfg.ClusterConfig.DependencySet {
```

Comment:
We are going to need to use `slices.SortFunc` from the go standard library because the iteration over a map in go is purposefully not deterministic